test/unit/coffee/ChannelManagerTests.coffee

```diff
@@ -33,26 +33,53 @@ describe 'ChannelManager', ->
 			it "should not subscribe to the redis channel", ->
 				@rclient.subscribe.called.should.equal false
 
-		describe "when subscribe errors", ->
+		describe "with unhandledRejection warnings suppressed", ->
 			# TODO(das7pad): rework after decaff -- our coffee-script version does not like async/await
-			beforeEach (done) ->
-				@rclient.subscribe = () ->
-					return new Promise (resolve, reject) ->
-						setTimeout((() -> reject(new Error("some redis error"))), 1)
-				p = @ChannelManager.subscribe @rclient, "applied-ops", "1234567890abcdef"
-				@rclient.subscribe = sinon.stub().resolves()
-				p.then () ->
-					done(new Error('should not subscribe but fail'))
-				p.catch (err) =>
```

Comment:
I think the unhandled rejection is coming from the p.then(), which returns a promise which is not handled.

If you change it to 

```
p.then () ->
	done(new Error('should not subscribe but fail'))
.catch (err) =>
	err.message.should.equal "some redis error"
        ....
```
Then the warning will go away and you don't need the changes in the first commit.

Reply:
OK, so we have to handle rejects on every attached promise. This is unexpected at first, but makes sense in that some arbitrary handler should always handle success and failure. (This may have catched #134.)

<details>

<pre>
> reject=0; p = new Promise((_, r) => {reject=r})
Promise { <pending> }
> p.catch((reason)=>console.log({reason}))
Promise { <pending> }
> p.then()
Promise { <pending> }
> p.then()
Promise { <pending> }
> p.then()
Promise { <pending> }
> reject('boom')
undefined
> { reason: 'boom' }
(node:19102) UnhandledPromiseRejectionWarning: boom
(node:19102) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:19102) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:19102) UnhandledPromiseRejectionWarning: boom
(node:19102) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:19102) UnhandledPromiseRejectionWarning: boom
(node:19102) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
</pre>
</details>

Reply:
Done and force pushed.